.github/workflows/ci-github-actions-self-hosted.yaml

```diff
@@ -7,7 +7,12 @@ on:
 
 jobs:
   gpu-cuda:
-    if: github.repository_owner == 'QMCPACK'
+    if: |
+        github.repository_owner == 'QMCPACK' &&
+        github.event.issue.pull_request &&
+        ( startsWith(github.event.comment.body, 'Test this please') || 
+          startsWith(github.event.comment.body, 'Start testing in-house') )
```

Comment:
Does "startsWith" mean a comment starts with the given phrase
or testing starts from the last commit and applies to all subsequent commits?

It looks like the former. github action is new to me, so I got confused.

If I understand correctly, With this PR, 'Start testing in-house' will behave exactly the same as 'Test this please' on the github action side CI testing and we don't have something similar to "Okay to test" on Jenkins side as I described in https://github.com/QMCPACK/qmcpack/pull/3260#issuecomment-870886859.

Reply:
Yes, it's the former. Just to make sure the string starts with it as that the intention, since a comment can contain those phrases in a different context. 

> If I understand correctly, With this PR, 'Start testing in-house' will behave exactly the same as 'Test this please' on the github action side CI testing and we don't have something similar to "Okay to test" on Jenkins side as I described in #3260 (comment).

Yes, for now it's behaving that way. Let me see if there is a way to replicate "Okay to test" with the GitHub Actions API (check if status exists?).